From 40c9939151bb56b1ed6e4c6cddc40f2a3bb5e882 Mon Sep 17 00:00:00 2001
From: Francois Daoust <fd@tidoust.net>
Date: Thu, 30 Sep 2021 23:00:09 +0200
Subject: [PATCH] Drop duplicate interfaces, dictionaries and properties

The terms are partially re-defined in Level 2. Patch is only needed because
Level 2 is a delta spec. It will likely need to be kept around for as long
as that remains the case.
---
 ed/idl/web-animations.idl | 18 ------------------
 1 file changed, 18 deletions(-)

diff --git a/ed/idl/web-animations.idl b/ed/idl/web-animations.idl
index 6e973d195..2b5f672b9 100644
--- a/ed/idl/web-animations.idl
+++ b/ed/idl/web-animations.idl
@@ -25,8 +25,6 @@ interface Animation : EventTarget {
              attribute DOMString                id;
              attribute AnimationEffect?         effect;
              attribute AnimationTimeline?       timeline;
-             attribute double?                  startTime;
-             attribute double?                  currentTime;
              attribute double                   playbackRate;
     readonly attribute AnimationPlayState       playState;
     readonly attribute AnimationReplaceState    replaceState;
@@ -65,7 +63,6 @@ dictionary EffectTiming {
     FillMode                           fill = "auto";
     double                             iterationStart = 0.0;
     unrestricted double                iterations = 1.0;
-    (unrestricted double or DOMString) duration = "auto";
     PlaybackDirection                  direction = "normal";
     DOMString                          easing = "linear";
 };
@@ -86,10 +83,7 @@ enum FillMode { "none", "forwards", "backwards", "both", "auto" };
 enum PlaybackDirection { "normal", "reverse", "alternate", "alternate-reverse" };
 
 dictionary ComputedEffectTiming : EffectTiming {
-    unrestricted double  endTime;
-    unrestricted double  activeDuration;
-    double?              localTime;
     double?              progress;
     unrestricted double? currentIteration;
 };
 
@@ -158,14 +152,3 @@ partial interface mixin DocumentOrShadowRoot {
 };
 
 Element includes Animatable;
-
-[Exposed=Window]
-interface AnimationPlaybackEvent : Event {
-    constructor(DOMString type, optional AnimationPlaybackEventInit eventInitDict = {});
-    readonly attribute double? currentTime;
-    readonly attribute double? timelineTime;
-};
-dictionary AnimationPlaybackEventInit : EventInit {
-    double? currentTime = null;
-    double? timelineTime = null;
-};
-- 
2.31.1.windows.1

